docs/api.md

```diff
@@ -929,6 +929,35 @@ To submit a new translation for yargs:
 <a name="nargs"></a>.nargs(key, count)
 -----------
 
+.middleware(callbacks)
+------------------------------------
+
+Define global middleware functions to be called first, in list order, for all cli command.  
+
+The `callbacks` parameter can be a function or a list of functions.  
+
+```js
+const mwFunc1 = argv => console.log('I\'m a middleware function');
+const mwFunc2 = argv => console.log('I\'m another middleware function');
+
+yargs
+  .command('myCommand', 'some command', {}, function(argv){
+    console.log('Running myCommand!');
+  })
+  .middleware([mwFunc1, mwFunc2]).argv;
```

Comment:
can I pass in a single function, rather than an array?

Reply:
Yea, it can either be a single function or a list of functions.